### PR TITLE
add efficientnetv2

### PR DESCRIPTION
**Pull request by xiaohu2015**
I want to add efficientnetv2. https://github.com/pytorch/vision/issues/2707

### CODE REVIEW

**Comment by xiaohu2015** [comment]
I implement EfficientNetV2 just like the style of EfficientNetV1 in torchvision models. I also converted the timm pretrained weights, but I found the acc dropped a lot (for example, EfficientNetV2-S is about 82.9 rather 83.9) because of the different padding for 3x3 conv of stride2.

**Comment by datumbox** [comment]
@xiaohu2015 First of all let me confirm that the EfficientNetV2 is within the list of models we would like to include on TorchVision. So the contribution is definitely useful and thank you for it. Moreover your code is of high quality and follows the conventions/style of TorchVision.

Unfortunately accepting model contributions from the community has been challenging in the past, primarily due to training issues. You can read more about this at https://github.com/pytorch/vision/issues/2707#issuecomment-937581682. The TLDR for it is that we don't accept contributions without trained weights which fully reproduce the accuracy of the paper. On the one hand, getting the weights is quite a lot to ask from open-source contributors because training is quite expensive. On the other, a contribution without weights means that an internal maintainer needs to do a significant amount of work to reproduce the paper and that might not be possible due to different priorities. Taking weights from elsewhere is something done in the past but we usually try to avoid for various reasons (lack of reproducibility, minor implementation differences can lead to large accuracy gaps as the ones you report above, adds little value to the community etc).

This is why especially for very large features such as this, it's imperative that we discuss the details prior starting the work. Here are a few things we want to typically clarify:
1. Confirm that the feature is needed 
2. Find one maintainer from our team with the capacity to review/verify/guide the effort
3. Discuss the details of the implementation with the maintainer and agree on the direction

Doing the above help us identify the risks, plan features based on our capacity and ensure we don't throwaway work. I've been an Open-source contributor myself in the past and there is nothing worst than spending couple of weeks to write a feature only to be rejected or to remain unmerged for months due to lack of priority... This is the reason we want to clarify all the above prior starting writing the PR.

Concerning the contents of this PR, here are a few concerns:
1. Though I see you try to inherit components from EfficientNet, there is a lot of code duplication. Ideally we would like to extend V1 in a BC way to accommodate most of the changes of V2. Where we plan to extend V1 and where we introduce a new class for V2 is non-trivial and requires investigation.
2. The PR needs to be accompanied with pre-trained weights, trained from scratch using our reference scripts and fully reproduce the performances of the paper. That is only possible if you have access on hardware that will allow you to train the model.

I think to make this PR mergeable will require a lot of work, so I would like to understand the situation on your side. :)

**Comment by xiaohu2015** [comment]
@datumbox thanks, I am looking foward to your implemention. I will close this pr.

**Comment by datumbox** [comment]
@xiaohu2015 Your PRs are definitely welcome. If you have any other features you would like to work on let's chat about it on their linked issues to map out the details. :)

**Comment by xiaohu2015** [comment]
@datumbox Ok, thanks again.

**Comment by bratao** [comment]
Thank you @xiaohu2015 for this great implementation!

**Comment by xiaohu2015** [comment]
@bratao you can get the original implementation here: https://github.com/xiaohu2015/nncls/blob/main/models/efficientnetv2.py

**Comment by datumbox** [comment]
@xiaohu2015 We should consider adapting the existing efficientnet.py implementation of TorchVision to support both V1 and V2 and port it in. This can wait for after your other PR #4961 about FCOS is merged. :)
